### DISCUSSION TITLE
FDS and CFD Model Limitations? Can it model a gas leak coming from a pressurized cylinder that is overfilled?

### DISCUSSION THREAD

**Discussion by PapaJoe75** [firemodels/fds#13267] [General]
Can FDS model a propane leak from an LPG cylinder that has been overfilled and is being bled into the atmosphere intentionally in the interior of a building?  I am trying to determine the gas concentration near the cylinder inside the space (i.e. compartment) that it is released.

The internal pressure of the cylinder is 100-200 PSI (depending on the temperature). The intent of bleeding the LPG from the bleed valve is to bring the cylinder down to 80 percent by volume because the cylinder was overfilled to begin with.

Are the physics within FDS (subsonic) limitation and is there a validation study that verifies that FDS can account for the internal pressure of a gas cylinder and the external pressure of the atmosphere that is substantially less?  That's 2 questions.

How do you account for the differences in pressure in FDS (or can you) given the fact that LPG is being released in a vapor from the bleed valve (whose momentum and direction are questionable) and which quickly becomes a gas that occupies approximately 270 times the volume that it did inside the cylinder?

My information is that the hydrodynamic model within FDS does not consider or cover " incompressibility."

Joe

P.S. The bleed valve of an LPG cylinder has more of a horizontal than vertical momentum. How would FDS account to its direction and speed which will affect its migration and concentration?

**Comment by rmcdermo** [comment]
You don't need CFD to model the rate of release.  There are simple algebraic models for compressible orifice flow.  This will give you the mass loss rate of propane from the tank (it will be function of time).  From there you can use FDS (with reasonable resolution) to model the concentrations.  Let me know if you can't find the orifice flow equations, I can dig them up.

**Reply by PapaJoe75** [reply]
Thanks Randy....Let me search of the orfice flow equations. I am aware of this approach.

The question is "Is FDS capable of calculating (i.e. predicting the mass loss rate or amount of gas) as well as calculating and displaying the  direction and amount of gas release given the pressure differences with the interior of the cylinder vs the exterior to which it is released.

Quite simply, is this within its limitation? I suspect the gas release through the value exceeds its subsonic limitation or does that apply only to heat and smoke from a fire? I suspect the momentum of the leak through the bleed valve exceeds that.

**Reply by rmcdermo** [reply]
As Jason mentioned, at 200 psi you are choked.  FDS cannot compute the right discharge flow rate, even if you could somehow resolve the orifice (which you can't).  Directionally, if the flow were low-Mach, then FDS would do a reasonable job of predicting the flow rate (again with enough resolution, like 20 cells across the orifice, which would be modeled as a square).  The direction would be determined by your orientation of the OBST use and would be restricted to the Cartesian coordinate directions.

In your case, what I would do is to locate a particle next to your "tank" and from there you can discharge a flow (with a RAMP) and give it a specific orientation (flow direction with velocity components).

**Reply by PapaJoe75** [reply]
Can you locate and send me the orifice flow equations to calculate the mass flow of LPG from a bleed valve associated with an overfilled forklift cylinder?  I can't seem to find them.....Joe

**Reply by drjfloyd** [reply]
https://en.wikipedia.org/wiki/Choked_flow

**Comment by drjfloyd** [comment]
You would have to specify the leak rate. This is not something FDS is going to predict for you. 
FDS is a low Mach number code (Mach ~0.3 or less).  If your leak rate exceed this, then you will see a higher uncertainty in regions where the flow speed is not low speed. There are correlations in the literature for the expansion of sonic jets that you could use to estimate the size of the region before entrainment drops the speed. If that region is a large fraction of your region of interest, then FDS may not be the correct tool for this application.

**Comment by PapaJoe75** [comment]
Thank both of you for your analysis and observations. Your knowledge  confirms what I already suspected  and points me in the right direction of solving the problem. Are you ware of any modeling program that can more effectively deal with the issue.

**Reply by rmcdermo** [reply]
As an engineer, you do not want to use a CFD model for computing the discharge.  What you need are the orifice flow equations and  data from the manufacturer of the valve.  The coefficient is likely between 0.6 and 0.85.  Once you know the flow rate (range of possible flow rates), then FDS can show you where the propane will go.  Or you can use any other CFD code (FLUENT, OpenFOAM, etc.).
